src/update/UpdateManager.cpp

```diff
@@ -0,0 +1,150 @@
+/*!********************************************************************
+ Audacity: A Digital Audio Editor
+
+ @file UpdateManager.cpp
+ @brief Declare a class that managing of updates.
+
+ Anton Gerasimov
+ **********************************************************************/
+
+#include "UpdateManager.h"
+#include "UpdatePopupDialog.h"
+
+#include "NetworkManager.h"
+#include "IResponse.h"
+#include "Request.h"
+
+#include "widgets/AudacityMessageBox.h"
+
+#include <wx/platinfo.h>
+#include <wx/utils.h>
+
+static const char* prefsUpdatePopupDialogShown = "/Update/UpdatePopupDialogShown";
+static const char* prefsUpdateScheduledTime = "/Update/UpdateScheduledTime";
+
+enum { ID_TIMER = wxID_HIGHEST + 1 };
+
+BEGIN_EVENT_TABLE(UpdateManager, wxEvtHandler)
+    EVT_TIMER(ID_TIMER, UpdateManager::OnTimer)
+END_EVENT_TABLE()
+
+UpdateManager::UpdateManager(AudacityProject& project)
+    : mTrackingInterval(
+        std::chrono::milliseconds(std::chrono::hours(12)).count())
+{
+    mParent = (wxWindow*)(&GetProjectFrame(project));
+    wxASSERT(mParent);
+
+    mTimer.SetOwner(this, ID_TIMER);
+    mTimer.StartOnce();
+}
+
+UpdateManager::~UpdateManager()
+{
+    mTimer.Stop();
+}
+
+void UpdateManager::enableTracking(bool enable)
+{
+    gPrefs->Write(prefsUpdatePopupDialogShown, enable);
+    gPrefs->Flush();
+}
+
+bool UpdateManager::isTrackingEnabled()
+{
+    return gPrefs->ReadBool(prefsUpdatePopupDialogShown, true);
+}
+
+VersionPatch UpdateManager::getVersionPatch() const
+{
+    return mVersionPatch;
+}
+
+void UpdateManager::getUpdates()
+{
+    audacity::network_manager::Request request("https://updates.audacityteam.org/feed/latest.xml");
+    auto response = audacity::network_manager::NetworkManager::GetInstance().doGet(request);
+
+    response->setRequestFinishedCallback([response, this](audacity::network_manager::IResponse*) {
+
+        if (response->getError() != audacity::network_manager::NetworkError::NoError)
+        {
+            AudacityMessageBox(
+                XO("Unable to connect to Audacity update server."),
```

Comment:
As I also explained to Vitaly:

See the file FutureStrings.h.  Cut the strings from there.  Paste them into your source code.

Make sure you have copied them exactly, and also cut and pasted any i18n-hint comments that were immediately before the string.

The deletions from FutureStrings.h should also be committed to your branch.

Reply:
The strings in FutureStrings.h for this file all use XC (with two arguments), not XO.  It is important to use both arguments for correct lookup.